remote: OSF docs (#1958)
iterative/dvc.org: content/docs/command-reference/remote/add.md

<details>

### Click for OSF

An [Open Science Framework](https://osf.io/) account is needed to use

> OSF storage. Please also run `pip install osfclient` to install the Python

Reviewer: Is there good link to point "OSF storage" to? I looked in osf.io and cos.io and the best I could find is https://help.osf.io/hc/en-us/articles/360054528874 which isn't that great (doesn't explain what OSF storage is or how it works, etc.)

Reviewer: Per https://github.com/iterative/dvc/pull/4860#pullrequestreview-536210649 this paragraph doesn't seem to be accurate, actually?

Author: I agree that https://help.osf.io/hc/en-us/articles/360054528874 fits best here.

> An [OSF](https://osf.io/) account is needed to use OSF storage. Please also
> run `pip install osfclient` to install the Python package that allows DVC to
> communicate with the OSF API.

Reviewer: ```suggestion
An [Open Science Framework](https://osf.io/) account is needed to use OSF
storage. Please also run `pip install osfclient` to install the Python package that
allows DVC to communicate with the OSF API.
```

> run `pip install osfclient` to install the necessary Python package that enables
> DVC to communicate with the OSF API.

Reviewer: ```suggestion
run `pip install osfclient` to install the Python package that allows DVC to
communicate with the OSF API.
```

> Before start using [OSF](https://osf.io/) storage you should create osf aссount.
> After registration you also need run `pip install osfclient` to install package
> for working with OSF API. Finally, to add OSF remote, run

Reviewer: ```suggestion
An [OSF](https://osf.io/) account is needed to use OSF storage. Please also
run `pip install osfclient` to install the necessary Python package that enables
DVC to communicate with the OSF API.
```

<details>

### Click for OSF

An [Open Science Framework](https://www.cos.io/products/osf) account is needed to use
[OSF storage](https://help.osf.io/hc/en-us/articles/360054528874). 

```dvc
dvc remote add myremote osf://osf.io/path/to/dir
```
To work with OSF you need to specify several parameters:
- user
- project

> - password

Reviewer: Let's include the `remote modify` commands to add the minimum required credentials instead of the text under the code block, please.

Reviewer: I guess like this:
````suggestion
$ dvc remote add myremote osf://osf.io/path/to/dir
$ dvc remote modify osfremote --local project myproject
$ dvc remote modify osfremote --local user example@mail.com
$ dvc remote modify osfremote --local password 12345
```

> The project name, username, and password (may) contain sensitive user info.
> Therefore, it's safer to add them with the `--local` option, so they're
> written to a Git-ignored config file.
````
> Plus added the --local note like in the existing examples.

> See `dvc remote modify` for details.

Reviewer: ```suggestion
> See `dvc remote modify` for a full list of OSF parameters.
```

<details>

### Click for OSF

> An [Open Science Framework](https://www.cos.io/products/osf) account is needed to use
> [OSF storage](https://help.osf.io/hc/en-us/articles/360054528874). 

Reviewer: ```suggestion
An [Open Science Framework](https://osf.io/) account is needed to use
[OSF storage](https://help.osf.io/hc/en-us/articles/360054528874). 
```
